### PR TITLE
[#201] Core data structure with custom partitioning implemented.

### PR DESCRIPTION
**Pull request by mfbalin**

### CODE REVIEW

**Comment by mfbalin** [comment]
@VitoCastellana friendly ping

File: include/shad/core/vector.h [hunk 1]
```diff
@@ -0,0 +1,1044 @@
...
+
+#include "shad/data_structures/abstract_data_structure.h"
+#include "shad/distributed_iterator_traits.h"
+#include "shad/runtime/runtime.h"
+
+namespace shad {
+
+namespace impl {
+/// @brief Distributed vector without resizing capabilities.
+/// TODO change the descriptions
```
Comment (VitoCastellana) [inline include/shad/core/vector.h:38]: Please add the missing ref (or remove the comment Section XX)

**Review by VitoCastellana — APPROVED** [review]
Please just make the quick fix on the comment, then we are ready to merge! ;)
Thanks!! ;)

File: include/shad/core/vector.h (hunk 1 above)
Comment (mfbalin) [reply include/shad/core/vector.h:38]: Done

**Pull request merged**